templates/encrypt.nginx.tmpl

```diff
@@ -10,11 +10,27 @@ server {
     listen [::]:443 ssl http2 default_server;
     ssl_certificate {{fullchain}};
     ssl_certificate_key {{privkey}};
+    ssl_dhparam {{dhparam}};
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    ssl_prefer_server_ciphers on;
+    ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+    ssl_ecdh_curve secp384r1;
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_timeout 24h;
+    ssl_buffer_size 1400;
+    ssl_session_tickets off;
+    ssl_stapling on;
+    ssl_stapling_verify on;
+    resolver 8.8.8.8 8.8.4.4 valid=300s;
```

Comment:
If `resolver` is left blank, is the machine's default dns server used? Using the machine's default dns server is recommended, this is supplied by the cloud (MAAS ..) and/or might be changed by another charm such as landscape.